x-pack/legacy/plugins/uptime/public/state/api/utils.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpFetchQuery, HttpSetup } from '../../../../../../../target/types/core/public';
+
+class ApiService {
+  private static instance: ApiService;
+  public http: HttpSetup;
+
+  public set http(httpSetup: HttpSetup) {
+    this.http = httpSetup;
+  }
+
+  private constructor() {}
+
+  static getInstance(): ActionsBus {
+    if (!ApiService.instance) {
+      ApiService.instance = new ApiService();
+    }
+
+    return ApiService.instance;
+  }
+
+  public async get(apiUrl: string, params: HttpFetchQuery) {
+    const response = await this.http!.get(apiUrl, { query: params });
+    if (response instanceof Error) {
```

Comment:
Are we trying to make Kibana's fetch more like `window.fetch` where errors are thrown instead of returned?

Reply:
have updated this.